src/include/gcc.h

```diff
@@ -242,11 +242,11 @@ WT_ATOMIC_FUNC(size, size_t, size_t *vp, size_t v)
     } while (0)
 #define WT_READ_BARRIER()                           \
     do {                                            \
-        __asm__ volatile("dsb ishld" ::: "memory"); \
+        __asm__ volatile("dmb ishld" ::: "memory"); \
```

Comment:
The comment is actually correct, it says use ```dmb``` and not ```dsb```, the code diverged from the comment which this change fixes now. Do you mean it is more confusing now because we talk about ```dsb``` but then the code doesn't use any of it?

Reply:
I have updated the comment. I have added the context on what three barriers exist that should make it easier to understand why dmb is selected.